File: client/src/networking/packet/LogPacket.cpp
#include "LogPacket.h"

namespace Network { 
    namespace Packet {
        LogPacket::LogPacket(const String& message) : _logMessage(message) {

        }

        size_t LogPacket::estLength() {
            return this->_logMessage.length();
        }

        uint8_t LogPacket::id() {
            return 2;
        }

        void LogPacket::write(Buffer& buf) {
            buf.writeString(this->_logMessage);
        }

        void LogPacket::read(Buffer& buf) {
        }
    }
}

File: server/network/packet/LogPacket.go
package packet

import (
	"gardon.local/server/util"
	"io"
)

type LogPacket struct {
	Message string
}

func (pkt *LogPacket) ID() byte {
	return 2
}

func (pkt *LogPacket) Write(buf io.Writer) error {
	err := util.WriteString(buf, pkt.Message)
	return err
}

func (pkt *LogPacket) Read(buf io.Reader) error {
	readString, err := util.ReadString(buf)
	pkt.Message = readString
	return err
}
